terraso_backend/apps/soil_id/models/depth_dependent_soil_data.py

```diff
@@ -13,13 +13,298 @@
+        (SILT, _("Silt")),
+        (LOAM, _("Loam")),
+        (SANDY_CLAY_LOAM, _("Sandy clay loam")),
+        (SILTY_CLAY_LOAM, _("Silty clay loam")),
+        (CLAY_LOAM, _("Clay loam")),
+        (SANDY_CLAY, _("Sandy clay")),
+        (SILTY_CLAY, _("Silty clay")),
+        (CLAY, _("Clay")),
+    )
+    texture = models.CharField(null=True, choices=TEXTURES)
+
+    ROCK_FRAGMENT_0_1 = "0 - 1%"
+    ROCK_FRAGMENT_1_15 = "1 - 15%"
+    ROCK_FRAGMENT_15_35 = "15 - 35%"
+    ROCK_FRAGMENT_35_60 = "35 - 60%"
+    ROCK_FRAGMENT_60 = "> 60%"
+
+    ROCK_FRAGMENT_VOLUMES = (
+        (ROCK_FRAGMENT_0_1, _("0 - 1%")),
+        (ROCK_FRAGMENT_1_15, _("1 - 15%")),
+        (ROCK_FRAGMENT_15_35, _("15 - 35%")),
+        (ROCK_FRAGMENT_35_60, _("35 - 60%")),
+        (ROCK_FRAGMENT_60, _("> 60%")),
+    )
+    rock_fragment_volume = models.CharField(null=True, choices=ROCK_FRAGMENT_VOLUMES)
+
+    SUBSTEP_2_5 = "2.5"
+    SUBSTEP_5 = "5"
+    SUBSTEP_7_5 = "7.5"
+    SUBSTEP_10 = "10"
+
+    COLOR_HUES_SUBSTEPS = (
+        (SUBSTEP_2_5, _("2.5")),
+        (SUBSTEP_2_5, _("5")),
+        (SUBSTEP_7_5, _("7.5")),
+        (SUBSTEP_10, _("10")),
+    )
+    color_hue_substeps = models.CharField(null=True, choices=COLOR_HUES_SUBSTEPS)
+
+    RED = "R"
+    YELLOW_RED = "YR"
+    YELLOW = "Y"
+    GREEN_YELLOW = "GY"
+    GREEN = "G"
+    BLUE = "B"
+    BLUE_GREEN = "BG"
+
+    COLOR_HUES = (
+        (RED, _("r")),
+        (YELLOW_RED, _("yr")),
+        (YELLOW, _("y")),
+        (GREEN_YELLOW, _("gy")),
+        (GREEN, _("g")),
+        (BLUE, _("b")),
+        (BLUE_GREEN, _("bg")),
+    )
+    color_hue = models.CharField(null=True, choices=COLOR_HUES)
+
+    COLOR_VALUE_2_5 = "2.5"
+    COLOR_VALUE_3 = "3"
+    COLOR_VALUE_4 = "4"
+    COLOR_VALUE_5 = "5"
+    COLOR_VALUE_6 = "6"
+    COLOR_VALUE_7 = "7"
+    COLOR_VALUE_8 = "8"
+    COLOR_VALUE_8_5 = "8.5"
+    COLOR_VALUE_9 = "9"
+    COLOR_VALUE_9_5 = "9.5"
+
+    COLOR_VALUES = (
+        (COLOR_VALUE_2_5, _("2.5")),
+        (COLOR_VALUE_3, _("3")),
+        (COLOR_VALUE_4, _("4")),
+        (COLOR_VALUE_5, _("5")),
+        (COLOR_VALUE_6, _("6")),
+        (COLOR_VALUE_7, _("7")),
+        (COLOR_VALUE_8, _("8")),
+        (COLOR_VALUE_8_5, _("8.5")),
+        (COLOR_VALUE_9, _("9")),
+        (COLOR_VALUE_9_5, _("9.5")),
+    )
+    color_value = models.CharField(null=True, choices=COLOR_VALUES)
+
+    COLOR_CHROMA_1 = "1"
+    COLOR_CHROMA_2 = "2"
+    COLOR_CHROMA_3 = "3"
+    COLOR_CHROMA_4 = "4"
+    COLOR_CHROMA_5 = "5"
+    COLOR_CHROMA_6 = "6"
+    COLOR_CHROMA_7 = "7"
+    COLOR_CHROMA_8 = "8"
+
+    COLOR_CHROMAS = (
+        (COLOR_CHROMA_1, _("1")),
+        (COLOR_CHROMA_2, _("2")),
+        (COLOR_CHROMA_3, _("3")),
+        (COLOR_CHROMA_4, _("4")),
+        (COLOR_CHROMA_5, _("5")),
+        (COLOR_CHROMA_6, _("6")),
+        (COLOR_CHROMA_7, _("7")),
+        (COLOR_CHROMA_8, _("8")),
+    )
+    color_chroma = models.CharField(null=True, choices=COLOR_CHROMAS)
```

Comment:
question: why is this an enum instead of an int with min 1 and max 8?

Reply:
@paulschreiber I think with a small fixed set of integers like this, enums are a better choice since it ensures you can't accidentally assign incorrect or unsupported values to the field. With validators, you'd need additional checks to ensure only the allowed range of values is accepted and it only raises a [ValidationError] if the input doesn’t meet the criteria

Reply:
Let's store this as an integer field so we can do aggregate operations (sum, average, etc.).